File: Suma_impar.py
print ("Ingrese la cantidad de casos a evaluar", " ")
print ("1. Uno", " ")
print ("2. Dos", " ")
casos = int (input ("Respuesta: "))
print (" ")

def impares (n1, n2):
    impar = 0
    for i in range (n1, n2+1):
        if i%2 != 0:
            impar = impar + i
    return impar

def impares2 (n3, n4):
    impar2 = 0
    for i in range (n3, n4+1):
        if i%2 != 0:
            impar2 = impar2 + i
    return (impar2)

if casos == 1:
    print ("Ingrese el rango del caso a evaluar", " ")
    n1 = int (input("Inicio de rango: "))
    n2 = int (input("Fin de rango: "))
    print (" ")
    print ("La suma de todos los numeros impares en el rango ingresado es:", impares (n1, n2))

elif casos == 2:
    print ("Ingrese el rango del primer caso a evaluar", " ")
    n1 = int (input("Inicio de rango: "))
    n2 = int (input("Fin de rango: "))
    print (" ")
    print ("Ingrese el rango del segundo caso a evaluar", " ")
    n3 = int (input("Inicio de rango: "))
    n4 = int (input("Fin de rango: "))
    print (" ")
    print ("La suma de todos los numeros impares del primer caso ingresado es:", impares (n1, n2))
    print ("La suma de todos los numeros impares del segundo caso ingresado es:", impares2 (n3, n4))

else: print ("Valor ingresado es incorrecto")

File: Suma_impar.cpp
#include <iostream>
#include <cmath>

using namespace std;

int casos = 0, inicio = 0, inicio2 = 0, fin = 0, fin2 = 0;
int impar (int inicio, int fin);
int impar2 (int inicio2, int fin2);

int main (){
    cout << "Ingrese la cantidad de casos a evaluar" << endl;
    cout << "1. Uno" << endl << "2. Dos" << endl << "Respuesta: "; cin >> casos;
    cout << endl;

    switch (casos){
        case 1:
        cout << "Ingrese el rango del caso a evaluar" << endl;
        cout << "Inicio de rango: "; cin >> inicio;
        cout << "Fin de rango: "; cin >> fin;
        cout << endl;
        cout << "La suma de todos los numeros impares del caso ingresado es: " << impar (inicio,fin);
        break;

        case 2:
        cout << "Ingrese el rango del primer caso a evaluar" << endl;
        cout << "Inicio de rango: "; cin >> inicio;
        cout << "Fin de rango: "; cin >> fin;
        cout << endl;

        cout << "Ingrese el rango del segundo caso a evaluar" << endl;
        cout << "Inicio de rango: "; cin >> inicio2;
        cout << "Fin de rango: "; cin >> fin2;
        cout << endl;

        cout << "La suma de todos los numeros impares del primer caso ingresado es: " << impar (inicio,fin) << endl;
        cout << "La suma de todos los numeros impares del segundo caso ingresado es: " << impar2 (inicio2,fin2);
        break;

    }

    
    return 0;
}

int impar (int inicio, int fin){
    int impares = 0, i = 0;
    for (i = inicio; i <= fin; i++){
        if (i%2 != 0){
            impares = impares + i;
            }
    }

    return impares;
}

int impar2 (int inicio2, int fin2){
    int impares = 0, i = 0;
    for (i=inicio2; i<=fin2; i += 1){
        if (i%2 != 0){
            impares = impares + i;
        }
    }

    return impares;
}